pallets/pallet-did-lookup/src/lib.rs

```diff
@@ -334,6 +353,37 @@ pub mod pallet {
 
 			LinkableAccountDepositCollector::<T>::update_deposit(&account)
 		}
+
+		/// Executes the key type migration of the `ConnectedDids` and
+		/// `ConnectedAccounts` storages by converting the given `AccoundId`
+		/// into `LinkableAccountId(AccountId)`. Once all keys have been
+		/// migrated, the call `try-finalize-migration` will succeed and remove
+		/// the migration flag.
+		///
+		/// Can be called by any origin as duplicate calls won't succeed.
+		#[pallet::weight(<T as Config>::WeightInfo::migrate(*limit))]
+		#[pallet::call_index(254)]
+		pub fn migrate(origin: OriginFor<T>, limit: u32) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let migration_state = MigrationStateStore::<T>::get();
+
+			let new_last_key = match migration_state {
+				MigrationState::Done => None,
+				MigrationState::PreUpgrade => crate::migrations::do_migrate::<T>(limit, None)?,
+				MigrationState::Upgrading(last_key) => crate::migrations::do_migrate::<T>(limit, Some(last_key))?,
+			};
+
+			if let Some(migrated_acc) = new_last_key {
+				MigrationStateStore::<T>::set(MigrationState::Upgrading(migrated_acc));
+				Self::deposit_event(Event::<T>::MigrationProgress);
+			} else {
+				MigrationStateStore::<T>::set(MigrationState::Done);
```

Comment:
I now see that the filter prevents people from calling this extrinsic multiple times. It would then help to have a comment explaining that :)

Reply:
I honestly didn't put too much thought into this part. That didn't even notice that after the migration is done you can't call this call anymore.

That said I think it's not too bad that we would always send the `MigrationCompleted` event when someone calls the call. It's better than returning an error or doing nothing.
But I will add the comment and fix the doc 24e8ea006b74549732287aade3b95f2c931819f9

Reply:
Yes but this is not the case anymore. People calling the extrinsic after the migration is completed will get an error since the call will be filtered.

Reply:
yes that is true